typescript/sdk/src/providers/transactions/submitter/ethersV5/EV5GnosisSafeTxSubmitter.ts

```diff
@@ -0,0 +1,89 @@
+import SafeApiKit from '@safe-global/api-kit';
+import Safe, { EthSafeSignature } from '@safe-global/protocol-kit';
+import {
+  MetaTransactionData,
+  SafeTransactionData,
+} from '@safe-global/safe-core-sdk-types';
+import assert from 'assert';
+import { PopulatedTransaction } from 'ethers';
+import { Logger } from 'pino';
+
+import { Address, rootLogger } from '@hyperlane-xyz/utils';
+
+import { ChainName } from '../../../../types.js';
+import { getSafe, getSafeService } from '../../../../utils/gnosisSafe.js';
+import { MultiProvider } from '../../../MultiProvider.js';
+import { EV5Tx } from '../../TransactionTypes.js';
+import { TxSubmitterType } from '../TxSubmitterTypes.js';
+
+import { EV5TxSubmitterInterface } from './EV5TxSubmitterInterface.js';
+
+interface EV5GnosisSafeTxSubmitterProps {
+  safeAddress: Address;
+  signerAddress?: Address;
+}
+
+export class EV5GnosisSafeTxSubmitter implements EV5TxSubmitterInterface {
+  public readonly txSubmitterType: TxSubmitterType =
+    TxSubmitterType.GNOSIS_SAFE;
+
+  protected readonly logger: Logger = rootLogger.child({
+    module: 'gnosis-safe-submitter',
+  });
+
+  constructor(
+    public readonly multiProvider: MultiProvider,
+    public readonly chain: ChainName,
+    public readonly props: EV5GnosisSafeTxSubmitterProps,
+  ) {}
+
+  public async submit(...txs: EV5Tx[]): Promise<void> {
+    const safe: Safe.default = await getSafe(
+      this.chain,
+      this.multiProvider,
+      this.props.safeAddress,
+    );
+    const safeService: SafeApiKit.default = getSafeService(
+      this.chain,
+      this.multiProvider,
```

Comment:
I take back what I said about having a separate signer address, given that we attach the signer on the multiprovider in these utils